packages/utils/src/useBulkSelect/useBulkSelectWithItems.js

```diff
@@ -0,0 +1,48 @@
+import { useEffect, useMemo } from 'react';
+import { filterSelected } from '../useTableTools/helpers';
+import { selectItemTransformer, itemIds } from './helpers';
+import useBulkSelect from './useBulkSelect';
+
+const useBulkSelectWithItems = ({ onSelect, items: propItems, filter, paginator, preselected, setPage }) => {
+  const enableBulkSelect = !!onSelect;
+  const items = propItems.map((item) => selectItemTransformer(item, preselected));
+  const total = items.length;
+
+  const filteredItems = filter ? filter(items) : items;
```

Comment:
I'd simplify this by using null comparator
```suggestion
  const filteredItems = filter?.(items) ?? items;
```

Reply:
Added.